### ISSUE TITLE
support for third party tools (add datadog live tail link functionality to sam logs)

### ISSUE THREAD

**Issue by thedavlee** [aws/aws-sam-cli#3701]
### Describe your idea/feature/enhancement

We want to add automatic Datadog live tail link generation to sam logs.

The user flow would look something like this:
```
sam logs --stack-name david-samcli-stack --datadog-livetail       

This feature is currently in beta. Visit the docs page to learn more about the AWS Beta terms https://aws.amazon.com/service-terms/.
Enter Y to proceed with the command, or enter N to cancel:
 [y/N]: y
Experimental features are enabled for this session.
Visit the docs page to learn more about the AWS Beta terms https://aws.amazon.com/service-terms/.

If your functions are instrumented with Datadog, use the following links to see datadog live tailed logs:
HelloWorldFunction: https://app.datadoghq.com/logs/link-to-your-livetail
HelloWorld1Function: https://app.datadoghq.com/logs/link-to-your-livetail
```

### Proposal

Currently ([existing PR](https://github.com/aws/aws-sam-cli/pull/3697)) I have integrated the new code directly into sam logs. A possible plugin based model was mentioned?

Things to consider:
1. Will this require any updates to the [SAM Spec](https://github.com/awslabs/serverless-application-model)

**Comment by jfuss** [comment]
@thedavlee Thanks for the issue. It's much easier to iterate on thinking here before you spend time on a PR. 

So I see value to customers who are using Datadog to be able to stream their logs back, similar to what we do for CloudWatch. What I go back and forth on (personally) is: Does this belong directly in the CLI or is some kind of plugin better suited?

It's likely this is small so for us to maintain and handle bugs seems reasonable (or not a lot of effort) but feels like this can quickly become, "I have a different logging store and `sam logs` should read it". Now maybe the right thing is that we just support all, but I have seen this become never ending before. Maybe we need a criteria on what we shouldn't or shouldn't support directly in the CLI or just make it plug-able from the start. We don't have a plugin solution today, but we are thinking of one. 

I lean into the plugin solution, which could allow for streaming any log store back to the CLI. @praneetap What are your thoughts here?

**Comment by praneetap** [comment]
I agree, as we are working on a plugin-like solution on SAM CLI to support other IaC platforms, it could be extended to 3rd party log providers as well. It probably belongs there, so Datadog could be the first integration we do with sam logs command. I do want to hear from users on whether this is a solution that would help their development productivity, so 👍🏽 on this issue will help me prioritize.

**Comment by thedavlee** [comment]
Hi all, thanks for your thoughts. I did want to clarify that we're not currently planning to directly stream logs from Datadog and show them in the console (like what you do with CloudWatch), but rather print links to the Datadog app's live tail page for each given function. Does this change the plugin approach you all are envisioning?

**Comment by jfuss** [comment]
@thedavlee So the option just prints a link out for the customer to follow? If that is the case, I am not sure what the value is in adding this to SAM CLI. I would rather stream the logs. Only printing a link to follow doesn't seem very useful and starts to make the command act differently base on different inputs (making UX more complex and the customer have to jump around).

Even if we agreed that this was the right UX, I don't think that changes anything from the plugins. Mainly because we are still opening the door to support multiple different logging solutions rather than those be add ons.
